pkg/kubelet/prober/prober.go

```diff
@@ -142,9 +147,18 @@ func (pb *prober) runProbe(ctx context.Context, probeType probeType, p *v1.Probe
 		return pb.exec.Probe(pb.newExecInContainer(ctx, container, containerID, command, timeout))
 
 	case p.HTTPGet != nil:
-		req, err := httpprobe.NewRequestForHTTPGetAction(p.HTTPGet, &container, status.PodIP, "probe")
-		if err != nil {
-			return probe.Unknown, "", err
+		var err error
+		var req *http.Request
+		key := probeKey{podUID: pod.UID, containerName: container.Name, probeType: probeType}
+		reqAny, found := httpProbeReqCache.Load(key)
+		if !found || cachedRequestStaled(p, reqAny.(*http.Request), status) {
+			req, err = httpprobe.NewRequestForHTTPGetAction(p.HTTPGet, &container, status.PodIP, "probe")
+			if err != nil {
+				return probe.Unknown, "", err
+			}
+			httpProbeReqCache.Store(key, req)
+		} else {
+			req = reqAny.(*http.Request)
 		}
```

Comment:
```suggestion
                req = reqAny.(*http.Request)
		if !found || (p.HTTPGet.Host == "" && req.URL.Hostname() != status.PodIP) {
			req, err = httpprobe.NewRequestForHTTPGetAction(p.HTTPGet, &container, status.PodIP, "probe")
			if err != nil {
				return probe.Unknown, "", err
			}
			httpProbeReqCache.Store(key, req)
		}
```

Reply:
Sorry, as a newbie in go, I'm not sure what will happen in case of a cache miss, reqAny will be nil? Is it ok to do a type assertion in that case?

Reply:
Good question. Go will panic trying to convert nil interface to *http.Request, I guess. My bad.

Reply:
I've updated the suggestion.

Reply:
this is really neat. I'll make those changes

Reply:
Frankly, It doesn't look very readable to me. I'd even try to avoid using sync.Map because of this and use plain map + locks (if locks are really needed here).